contracts/contracts/strategies/LidoWithdrawalStrategy.sol

```diff
@@ -0,0 +1,314 @@
+ * @author Origin Protocol Inc
+ */
+contract LidoWithdrawalStrategy is InitializableAbstractStrategy {
+    /// @notice Address of the WETH token
+    IWETH9 private constant weth =
+        IWETH9(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+    /// @notice Address of the stETH token
+    IERC20 private constant stETH =
+        IERC20(0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84);
+    /// @notice Address of the Lido Withdrawal Queue contract
+    IStETHWithdrawal private constant withdrawalQueue =
+        IStETHWithdrawal(0x889edC2eDab5f40e902b864aD4d7AdE8E412F9B1);
+    /// @notice Maximum amount of stETH that can be withdrawn in a single request
+    uint256 public constant MaxWithdrawalAmount = 1000 ether;
+    /// @notice Total amount of stETH that has been requested to be withdrawn for ETH
+    uint256 public outstandingWithdrawals;
+
+    event WithdrawalRequests(uint256[] requestIds, uint256[] amounts);
+    event WithdrawalClaims(uint256[] requestIds, uint256 amount);
+
+    constructor(BaseStrategyConfig memory _stratConfig)
+        InitializableAbstractStrategy(_stratConfig)
+    {
+        require(MaxWithdrawalAmount < type(uint120).max);
+    }
+
+    /**
+     * @notice initialize function, to set up initial internal state
+     * @param _rewardTokenAddresses Address of reward token for platform
+     * @param _assets Addresses of initial supported assets
+     * @param _pTokens Platform Token corresponding addresses
+     */
+    function initialize(
+        address[] memory _rewardTokenAddresses,
+        address[] memory _assets,
+        address[] memory _pTokens
+    ) external onlyGovernor initializer {
+        InitializableAbstractStrategy._initialize(
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+        safeApproveAllTokens();
+    }
+
+    /**
+     * @notice deposit() function not used for this strategy. Use depositAll() instead.
+     */
+    function deposit(address, uint256) public override onlyVault nonReentrant {
+        // This method no longer used by the VaultAdmin, and we don't want it
+        // to be used by VaultCore.
+        require(false, "use depositAll() instead");
+    }
+
+    /**
+     * @notice Takes all given stETH and creates Lido withdrawal request
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 stETHStart = stETH.balanceOf(address(this));
+        require(stETHStart > 0, "No stETH to withdraw");
+
+        uint256 withdrawalLength = (stETHStart / MaxWithdrawalAmount) + 1;
+        uint256[] memory amounts = new uint256[](withdrawalLength);
+
+        uint256 stETHRemaining = stETHStart;
+        uint256 i = 0;
+        while (stETHRemaining > MaxWithdrawalAmount) {
+            amounts[i++] = MaxWithdrawalAmount;
+            stETHRemaining -= MaxWithdrawalAmount;
+        }
+        amounts[i] = stETHRemaining;
```

Comment:
Does this revert if the `stETHRemaining` == 0? Though considering current stETH balance on the vault is ` 17,886.18` this is highly unlikely to happen

Reply:
Confirmed it would revert since according to the docs (https://docs.lido.fi/contracts/withdrawal-queue-erc721/): 

_The minimal amount for a request is 100 wei, and the maximum is 1000 eth. More significant amounts should be split into several requests, which allows us to avoid clogging the queue with an extra large request._